### DISCUSSION TITLE
Multi-tenancy with multiple Models

### DISCUSSION THREAD

**Discussion by prykris** [filamentphp/filament#15066] [Help]
### Package

Panel builder

### Package Version

v3.2.128

### How can we help you?

My application has Teams, and I've set up my Filament to use it as a Tenant model. But I wish to scope the resources for a second model: The Site model.

User blongs to Team -> Team has Sites -> Sites have resources

Perhaps this issue could have been avoided if I had simply chosen to use Site as the tenant.

Most of those models don't belong directly to the Team directly but to the Site model and I am scoping them to the Team via this relationship

```php
public function team(): HasOneThrough
    {
        return $this->hasOneThrough(
            Team::class,
            Site::class,
            'id',
            'id',
            'site_id',
            'team_id'
        );
    }
``` 

I am not sure if I'm overthinking this or not this does seem like a standard practice where each Team can have multiple Sites and each of those resources are scoped to Site.

Even more annoying is the fact the those URLs constructed only have a single ID for Team component, and not the Site so I have to store it on Users table and apply tenant global scopes manually via middleware.

It gets more frustrating for Editing and Creating records since they are looking for those resource relationship on the Tenant (Team) model and have no way to set the ID for Site which is what is necessary in this case.

### Second question

The tenant selection is nice, but I wish to have something similar for Site selection, so there are two select dropdowns.

**Comment by thekirkulees** [comment]
+1 - Would use this to disable memory for specialized task models (e.g., portfolio assistant) while keeping it enabled for general assistant models.
